airflow/providers/apache/druid/CHANGELOG.rst

```diff
@@ -27,6 +27,19 @@
 Changelog
 ---------
 
+4.0.0
+.....
+
+Features
+~~~~~~~~
+
+* ``Add druid ingestion connection test And expose druid ingest hook (#33795)``
+
+.. note::
+  The connection type of Druid has been separated into two(druid_broker, druid_ingest)
```

Comment:
Please elaborate on this one. We should explain to users how to mitigate the issue (leave instructions how to migrate to this new version)

Reply:
@eladkal 
I'm not an English speaker, so I'm not sure if the following mention is accurate. Please review this

```
The connection type of Druid has been separated into two(Druid Broker, Druid Ingest). Please perform one of the two methods listed below.
  1. update druid connect type to (Druid Broker, Druid Ingest) on UI screen (Admin/Connections)
  2. run command airflow db migrate
```

**but in second case, airflow db migrate doesn't support update connect type** so we need implement this. 
my suggestion is write first option only, and add an implementation to check the version of the Druid provider in the db migrate process, so that we can upgrade the Druid connector type when needed in Airflow in other issue.

Reply:
Nope, we don't have connection migrations, and with current release model (that doesn't mean it is bad) I don't think it even possible to implement it. All migration happen only on upgrade Airflow Core, not Providers.

Just some examples of the past breaking changes and their descriptions:

1. Remove AWS S3 Connection (spoiler alert, it does impact some users)
https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/changelog.html#id66

2. Removed deprecated fields in Slack Provider (still wait when users starts complain)
https://airflow.apache.org/docs/apache-airflow-providers-slack/stable/changelog.html#breaking-changes

3. Deprecated parameter removed from Google Provider
https://airflow.apache.org/docs/apache-airflow-providers-google/stable/changelog.html#breaking-changes

The main point of this breaking changes. When user open issue/discussion in Github or write in slack it much easier to send them link to Breaking Changes, rather then try to figure out how to solve their issue.

Reply:
@eladkal @Taragolis 
Taragolis thanks for adding example link
in this pr the only concern point is changing connection type. So, I think this mention is enough.
```
In this version of provider Apache Druid Connection(conn_type="druid") has been separated into two(Druid Broker, Druid Ingest)
Update connect type(conn_type = "druid") to druid broker, druid ingest 
```

Reply:
The goal of the note is to explain what actions needed to be done in order to migrate.
The note you listed says the conn is split in two so:
1. How do I choose which one?
2. If I want to have my code work exactly as it is before what steps I need to do?

These are the two questions the notes must answer